proknow-sdk/Logs/Audit.cs

```diff
@@ -0,0 +1,220 @@
+        /// </summary>
+        /// <param name="proKnow">Parent ProKnow object</param>
+        internal Audit(ProKnowApi proKnow)
+        {
+            _proKnow = proKnow;
+        }
+
+        /// <summary>
+        /// Gets audit logs asynchronously
+        /// </summary>
+        /// <returns>A page of audit logs</returns>
+        /// <example>This example shows how to get the first page of audit logs:
+        /// <code>
+        /// using ProKnow;
+        /// using System.Threading.Tasks;
+        ///
+        /// var pk = new ProKnowApi("https://example.proknow.com", "./credentials.json");
+        /// FilterParameters filterParams = new FilterParameters();
+        /// var auditLogs = await _proKnow.Audit.Query(filterParams);
+        /// </code>
+        /// </example>
+        public async Task<AuditPage> Query(FilterParameters filter)
+        {
+            this.filterParameters.Copy(filter);
+
+            this.filterParameters.PageNumber = null;
+            if (filter == null)
+            {
+                this.filterParameters.PageSize = 25;
+            }
+
+            var bodyJson = JsonSerializer.Serialize(filterParameters, serializerOptions);
+            var requestContent = new StringContent(bodyJson, Encoding.UTF8, "application/json");
+
+            var json = await _proKnow.Requestor.PostAsync("audit/events/search", null, requestContent);
+            var page = JsonSerializer.Deserialize<AuditPage>(json);
+
+            if (page.Items.Count > 0)
+            {
+                this.filterParameters.FirstId = page.Items[0].Id;
+                this.filterParameters.PageNumber = 0;
+            }
+
+            return page;
+        }
+
+        /// <summary>
+        /// Gets next page of audit logs asynchronously
+        /// </summary>
+        /// <returns>The next page of audit logs</returns>
+        /// <example>This example shows how to get the next page of audit logs:
+        /// <code>
+        /// using ProKnow;
+        /// using System.Threading.Tasks;
+        ///
+        /// var pk = new ProKnowApi("https://example.proknow.com", "./credentials.json");
+        /// var auditLogs = await _proKnow.Audit.Next();
+        /// </code>
+        /// </example>
+        public async Task<AuditPage> Next()
+        {
+            if (this.filterParameters.FirstId == null)
+            {
+                throw new ProKnowException("Must call Query first");
+            }
+
+            ++this.filterParameters.PageNumber;
+
+            var bodyJson = JsonSerializer.Serialize( this.filterParameters, serializerOptions);
+            var requestContent = new StringContent(bodyJson, Encoding.UTF8, "application/json");
+
+            var json = await _proKnow.Requestor.PostAsync("audit/events/search", null, requestContent);
+            var auditItem = JsonSerializer.Deserialize<AuditPage>(json);
+
+            return auditItem;
+        }
+
+        private class FilterParametersExtended
```

Comment:
Please make the class FilterParametersExtended internal and move it to its own file.